controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -777,8 +813,34 @@ func (r *StorageConsumerReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		},
 	)
 
+	clientStatusPredicate := predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			return false
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			objOld := e.ObjectOld.(*ocsv1alpha1.StorageConsumer)
+			objNew := e.ObjectNew.(*ocsv1alpha1.StorageConsumer)
+			return objOld.Status.Client != nil && objNew.Status.Client == nil
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return false
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
+			return false
+		},
+	}
```

Comment:
Why do we care about this? This is the storage consumer controller you should be acting to ant storage consumer change

Reply:
That does not make sense at all, the storage consumer controller should react to any change to the storage consumer.  I think the reason you are not getting notification is because you applied `GeneraionChangePredicate` which is not something that make sense if you want to watch for status changes